src/main/scala/mesosphere/marathon/state/Group.scala

```diff
@@ -106,12 +106,11 @@ case class Group(
   def removeApplication(appId: PathId): Group = copy(apps = apps.filter(_.id != appId))
 
   def makeGroup(gid: PathId): Group = {
-    val restPath = gid.restOf(id)
-    if (gid.isEmpty || restPath.isEmpty) this //group already exists
+    if (gid.isEmpty) this //group already exists
     else {
-      val (change, remaining) = groups.partition(_.id.restOf(id).root == restPath.root)
-      val toUpdate = change.headOption.getOrElse(Group.empty.copy(id = id.append(restPath.rootPath)))
-      val nestedUpdate = if (restPath.isEmpty) toUpdate else toUpdate.makeGroup(restPath.child)
+      val (change, remaining) = groups.partition(_.id.restOf(id).root == gid.root)
+      val toUpdate = change.headOption.getOrElse(Group.empty.copy(id = id.append(gid.rootPath)))
+      val nestedUpdate = if (gid.isEmpty) toUpdate else toUpdate.makeGroup(gid.child)
```

Comment:
Thanks for pointing that out, I did miss that one. Your suggested snipped causes the equivalent problem with multiple names though. It is due to the `restOf` function, which compares the head of both PathId, cutting off relevant subgroups, which happen to have the same name. By the recursive nature of `makeGroup`, we loose information about the initial parent path with each recursion step. We need the initial path though, in order to distinguish the depth of the path and find the actual rest of value. Instead I would suggest following:
```scala
this.copy(groups = remaining + toUpdate.makeGroup(id.child))
```
So basically getting rid of the if statement. Through the recursion we will step into returning `this` anyway, so it should be equivalent to still having the if statement.